### PR TITLE
Better config

### PR DESCRIPTION
**Pull request by OskarStark**
Can be merged after
* https://github.com/php-llm/llm-chain-bundle/pull/24

### CODE REVIEW

File: src/DependencyInjection/Configuration.php [hunk 1]
```diff
@@ -47,19 +47,36 @@ public function getConfigTreeBuilder(): TreeBuilder
...
+//                    ->arrayNode('mongodb')
+//                        ->children()
+//                            ->scalarNode('engine')->defaultValue('mongodb')->end()
+//                            ->scalarNode('uri')->isRequired()->end()
+//                            ->scalarNode('database_name')->isRequired()->end()
+//                            ->scalarNode('collection_name')->isRequired()->end()
+//                            ->scalarNode('index_name')->end()
+//                            ->scalarNode('vector_field_name')->defaultValue('vector')->end()
+//                            ->booleanNode('bulk_write')->defaultFalse()->end()
+//                        ->end()
```
Comment (OskarStark) [inline src/DependencyInjection/Configuration.php:77]: This is much more maintainable to me

File: src/DependencyInjection/Configuration.php [hunk 2]
```diff
@@ -47,19 +47,36 @@ public function getConfigTreeBuilder(): TreeBuilder
...
                 ->end()
-                ->arrayNode('stores')
-                    ->normalizeKeys(false)
-                    ->useAttributeAsKey('name')
-                    ->arrayPrototype()
+            ->arrayNode('stores')
+                ->children()
+                    ->arrayNode('chroma_db')
+                        ->children()
+                            ->scalarNode('engine')->defaultValue('chroma-db')->end()
```
Comment (chr-hertel) [inline src/DependencyInjection/Configuration.php:54]: we could drop the engine key now, right?

Comment (OskarStark) [reply src/DependencyInjection/Configuration.php:54]: I think so, yes

Comment (OskarStark) [reply src/DependencyInjection/Configuration.php:54]: Or the best with config component 😄 feel free to test my PR or finish it 😊

**Comment by OskarStark** [comment]
This will not work, after this change I am not able to have:

```yaml
    stores:
        chroma_db_products:
            engine: 'chroma-db'
            collection_name: '%env(CHROMA_COLLECTION_PRODUCTS)%'
        chroma_db_foo:
            engine: 'chroma-db'
            collection_name: '%env(CHROMA_COLLECTION_FOO)%'
```

**Comment by chr-hertel** [comment]
true ...

**Pull request closed without merging**